From badfed69369dcaf68cb39dfe0747d9f4cf36c2a4 Mon Sep 17 00:00:00 2001
From: Barbora Simonova <bsmejkal@redhat.com>
Date: Tue, 30 Sep 2025 11:42:48 +0200
Subject: [PATCH] Issue 6753 - Port ticket tests

Description:
Add another sleep to prevent busy server on slow machine

Relates: https://github.com/389ds/389-ds-base/issues/6753

Reviewed by: ???
---
 dirsrvtests/tests/suites/plugins/attruniq_test.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/dirsrvtests/tests/suites/plugins/attruniq_test.py b/dirsrvtests/tests/suites/plugins/attruniq_test.py
index 51c6e0439..ad9589c91 100644
--- a/dirsrvtests/tests/suites/plugins/attruniq_test.py
+++ b/dirsrvtests/tests/suites/plugins/attruniq_test.py
@@ -756,6 +756,9 @@ def test_multiple_containers_add_across_subtrees(topology_st, attruniq, containe
     with pytest.raises(ldap.CONSTRAINT_VIOLATION):
         create_stage_user(topology_st, uid=STAGE_USER_2_CN, cn=[STAGE_USER_1_CN, STAGE_USER_2_CN])
 
+    # Inserting sleep to prevent repeatedly occuring 'Server is busy' errors
+    sleep(5)
+
     log.info('Verify that uniqueness rule is enforced')
     with pytest.raises(ldap.CONSTRAINT_VIOLATION):
         create_stage_user(topology_st, uid=STAGE_USER_2_CN, cn=[ACTIVE_USER_1_CN, STAGE_USER_2_CN])
@@ -1061,3 +1064,9 @@ def test_invalid_config_invalid_subtree(topology_st):
     _config_file(topology_st, action='restore')
     topology_st.standalone.restart()
 
+
+if __name__ == '__main__':
+    # Run isolated
+    # -s for DEBUG mode
+    CURRENT_FILE = os.path.realpath(__file__)
+    pytest.main("-s %s" % CURRENT_FILE)
-- 
2.52.0

